docs/topics/multiplatform/multiplatform-discover-project.md

```diff
@@ -1,169 +1,315 @@
+corresponding `jvmMain` source set that compiles to only the JVM. Kotlin allows using platform-specific dependencies in
+these source sets, for instance, JDK in `jvmMain`:
 
-        }
-        nativeMain {
+```kotlin
+// jvmMain/kotlin/jvm.kt
+// You can use Java dependencies in the `jvmMain` source set
+fun jvmGreeting() {
+    java.io.File("greeting.txt").writeText("Hello, Multiplatform!")
+}
+```
 
-        }
-        nativeTest {
+### Compilation to a specific target
 
-        }
-    }
+Compilation to a specific target works with multiple source sets. When Kotlin compiles a multiplatform project to a
+specific target, it collects all source sets labeled with this target and produces binaries from them.
+
+Consider an example with `jvm`, `iosArm64`, and `js` targets. Kotlin creates the `commonMain` source set for common code
+and the corresponding `jvmMain`, `iosArm64Main`, and `jsMain` source sets for specific targets:
+
+![Compilation to a specific target](specific-target-diagram.svg){width=700}
+
+During compilation to the JVM, Kotlin selects all source sets labeled with "JVM", namely, `jvmMain` and `commonMain`. It
+then compiles them together to the JVM class files:
+
+![Compilation to JVM](compilation-jvm-diagram.svg){width=700}
+
+Because Kotlin compiles `commonMain` and `jvmMain` together, the resulting binaries contain declarations from
+both `commonMain` and `jvmMain`. This is how you can choose which parts of your application should be shared.
+
+When working with multiplatform projects, remember:
+
+* If you want Kotlin to compile your code to a specific platform, declare a corresponding target.
+* To choose a directory or source file to store the code, first decide among which targets you want to share your code:
+
+    * If the code is shared among all targets, it should be declared in `commonMain`.
+    * If the code is used for only one target, it should be defined in a specific source set for that target (for
+      example, `jvmMain` for the JVM).
+* Code written in platform-specific source sets can access declarations from the common source set. For example, the
+  code in `jvmMain` can use code from `commonMain`. However, the opposite isn't true: `commonMain` can't use code
+  from `jvmMain`.
+* Code written in platform-specific source sets can use the corresponding platform dependencies. For example, the code
+  in `jvmMain` can use Java-only libraries, like [Guava](https://github.com/google/guava)
+  or [Spring](https://spring.io/).
+
+### Intermediate source sets
+
+In simple multiplatform projects, only common code is shared among all available platforms and platform-specific code.
+In practice, you often need more granular code sharing.
+
+Consider an example where you need to target all modern Apple devices and Android devices:
+
+```kotlin
+kotlin {
+    android()
+    iosArm64()         // 64-bit iPhone devices
+    macosArm64()   // Modern Apple Silicon-based Macs
```

Comment:
Comments are not vertically aligned, or is it the artifact of the .md preview in Github?